# Task 20 — Customer Plan Category

## Business Question

How many customers belong to each standardized subscription plan category?

## Analysis

Map the existing subscription statuses to standardized, human-readable categories using `replace()` while preserving the original status column.

## Code

In [1]:
import pandas as pd
import numpy as np
customers = pd.read_csv("starlink_customers.csv")
subscriptions = pd.read_csv("subscriptions.csv")
starlink = customers.merge(
    subscriptions,
    left_on = "unique_id",
    right_on = "customer_id",
    how = "left"
)
plan_mapping = {
    "Residential" : "Home",
    "Roam" : "Mobile",
    "Business" : "Commercial",
    "Mini" : "Budget"
}
starlink["plan_category"] = starlink["plan_name"].map(plan_mapping)
starlink_counted = starlink["plan_category"].value_counts().sort_values(
    ascending = False
)
print(starlink_counted)

plan_category
Home          5959
Mobile        2078
Commercial    1019
Budget         944
Name: count, dtype: int64


### Result

Home is the largest subscription category with 5,959 customers, followed by Mobile with 2,078 customers. Commercial accounts for 1,019 customers, while Budget has the smallest customer base with 944 customers.

### Business Insight

The customer base is concentrated primarily in the Home category, which represents almost 60% of all customers. Mobile is the second-largest category, while Commercial and Budget have significantly smaller customer bases.